src/RoboSaverVirtualModule.sol

```diff
@@ -181,24 +203,55 @@ contract RoboSaverVirtualModule {
     /// @param _amount The amount of $EURe to deposit or withdraw
     function adjustPool(PoolAction _action, uint256 _amount) external onlyKeeper {
         if (_action == PoolAction.WITHDRAW) {
-            _poolWithdrawal(CARD, _amount);
+            /// @dev Close the pool in case the $EURe available for withdrawal is less than the deficit
+            uint256 withdrawableEure =
+                BPT_STEUR_EURE.balanceOf(CARD) * BPT_STEUR_EURE.getRate() * (MAX_BPS - slippage) / MAX_BPS;
+            if (withdrawableEure < _amount) {
+                _poolClose(withdrawableEure);
+            } else {
+                _poolWithdrawal(_amount);
+            }
         } else if (_action == PoolAction.DEPOSIT) {
-            _poolDeposit(CARD, _amount);
+            _poolDeposit(_amount);
         }
     }
 
     /*//////////////////////////////////////////////////////////////////////////
                                    INTERNAL METHODS
     //////////////////////////////////////////////////////////////////////////*/
 
+    /// @notice Close the pool position by withdrawing all to $EURe
+    /// @param _minAmountOut The minimum amount of $EURe to withdraw from the pool
+    /// @return request_ The exit pool request as per Balancer's interface
+    function _poolClose(uint256 _minAmountOut) internal returns (IVault.ExitPoolRequest memory request_) {
+        /// @dev All asset related arrays should always follow this (alphabetical) order
+        IAsset[] memory assets = new IAsset[](3);
+        assets[0] = IAsset(address(STEUR));
+        assets[1] = IAsset(address(BPT_STEUR_EURE));
+        assets[2] = IAsset(address(EURE));
+
+        /// @dev Allow for one wei of slippage
+        uint256[] memory minAmountsOut = new uint256[](3);
+        minAmountsOut[2] = _minAmountOut - 1;
```

Comment:
already slipp. was accounted in the calc in L207, why to reduce 1 wei here?

Reply:
correct; ba8e040cf2f6265c2fc9fd2d45ac77e8c499ffde